From: Sage Weil <sage@inktank.com>
Date: Thu, 30 Aug 2012 16:26:16 -0600
Patch-mainline: 3.7
Git-commit: e209db7ace281ca347b1ac699bf1fb222eac03fe
References: FATE#312888
Subject: [PATCH] Btrfs: set journal_info in async trans commit worker

We expect current->journal_info to point to the trans handle we are
committing.

Signed-off-by: Sage Weil <sage@inktank.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/transaction.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/btrfs/transaction.c
+++ b/fs/btrfs/transaction.c
@@ -1221,6 +1221,8 @@ static void do_async_commit(struct work_
 	struct btrfs_async_commit *ac =
 		container_of(work, struct btrfs_async_commit, work.work);
 
+	current->journal_info = ac->newtrans;
+
 	btrfs_commit_transaction(ac->newtrans, ac->root);
 	kfree(ac);
 }
